handlers/list_student.go

```diff
@@ -0,0 +1,58 @@
+package handlers
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/go-openapi/runtime/middleware"
+
+	runtime "github.com/abdulmoeid7112/student-backend-portal"
+	domainErr "github.com/abdulmoeid7112/student-backend-portal/errors"
+	"github.com/abdulmoeid7112/student-backend-portal/gen/models"
+	"github.com/abdulmoeid7112/student-backend-portal/gen/restapi/operations"
+)
+
+// NewListStudent handles a request for retrieving students.
+func NewListStudent(rt *runtime.Runtime) operations.ListStudentsHandler {
+	return &listStudents{rt: rt}
+}
+
+type listStudents struct {
+	rt *runtime.Runtime
+}
+
+// Handle the list employees request.
+func (e *listStudents) Handle(params operations.ListStudentsParams) middleware.Responder {
+	mapInterface := make(map[string]interface{})
+	if params.Name!=nil {
+		mapInterface["name"] = aws.StringValue(params.Name)
+	}
+	if params.Age!=nil {
+		mapInterface["age"] = aws.Int64Value(params.Age)
+	}
+	if params.Level!=nil {
+		mapInterface["level"] = aws.StringValue(params.Level)
+	}
+	if params.Phone!=nil {
+		mapInterface["phone"] = aws.StringValue(params.Phone)
+	}
+	students, err := e.rt.Service().ListStudents(mapInterface, aws.Int64Value(params.Limit), aws.Int64Value(params.Offset))
+	if err != nil {
+		switch apiErr := err.(*domainErr.APIError); {
+		case apiErr.IsError(domainErr.NotFound):
+			return operations.NewGetStudentNotFound()
+		default:
+			return operations.NewUpdateStudentInternalServerError()
+		}
+	}
+	stdList := make([]*models.Student, 0)
+	for i := range students {
```

Comment:
for _, student := range students is better.. you wont need to access through index.